### PR TITLE
Dark Neutrino Generator

### PR DESCRIPTION
**Pull request by ikatza**
This an implementation of a exotic physics GENIE module of a Dark Neutrino: https://arxiv.org/abs/1807.09877

A presentation will be given to GENIE collaboration, people from the collaboration are invited to take a look.

### CODE REVIEW

File: config/EngelFormFactor.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+
+<alg_conf>
+
+  <param_set name="Default">
+
+  </param_set>
+  <!-- <param_set name="3H"> -->
+  <!--   <param type="int" name="DV-Nucleus"> 1000010030  </param> -->
```
Comment (mroda88) [inline config/EngelFormFactor.xml:22]: Clean up this rubbish

File: src/Physics/Decay/DarkSectorDecayer.cxx [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+    int pdg_code = mother.Pdg();
+    if(pdg_code == kPdgDNuMediator){
+      dcs = DarkMediatorDecayChannels();
+    }
+    else if(pdg_code == kPdgDarkNeutrino ||
+            pdg_code == kPdgAntiDarkNeutrino){
+      dcs = DarkNeutrinoDecayChannels(pdg_code);
+    }
+
+    // for ( const auto & dc : dcs ) {
```
Comment (mroda88) [inline src/Physics/Decay/DarkSectorDecayer.cxx:86]: Clean this rubbish

File: src/Physics/Decay/DarkSectorDecayer.cxx [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+  }
+  return dcs;
+}
+//____________________________________________________________________________
+void DarkSectorDecayer::SetSpaceTime(
+  std::vector<GHepParticle> & pp,
+  const GHepParticle & mother,
+  double total_amplitude) const
+{
+  // TODO DNU: pay attention to units being used in GENIE!
```
Comment (mroda88) [inline src/Physics/Decay/DarkSectorDecayer.cxx:273]: Remove the TODO line

**Review by mroda88 — COMMENTED** [review]
good, let's do the review and collect the rest of the comments

File: src/Physics/Decay/DarkSectorDecayer.cxx (hunk 2 above)
Comment (ikatza) [reply src/Physics/Decay/DarkSectorDecayer.cxx]: Done

File: src/Physics/Decay/DarkSectorDecayer.cxx (hunk 3 above)
Comment (ikatza) [reply src/Physics/Decay/DarkSectorDecayer.cxx]: Done by you

File: config/EngelFormFactor.xml (hunk 1 above)
Comment (ikatza) [reply config/EngelFormFactor.xml]: Done

**Comment by ikatza** [comment]
I have merged master into this branch and resolved the outstanding merge conflicts.

**Pull request merged**